### PR TITLE
Adding JPO-Deduplicator to jpo-utils

### PR DESCRIPTION
**Pull request by John-Wiens**

### CODE REVIEW

**Review by Michael7371 — CHANGES_REQUESTED** [review]
Can we configure these topologies to start based off of the same/similar environmental variables that the kafka-setup pod uses?

if `KAFKA_TOPIC_CREATE_ODE` then the raw ODE topologies are enabled.
if `KAFKA_TOPIC_CREATE_GEOJSONCONVERTER` then the Processed topologies are enabled.
if `KAFKA_TOPIC_CREATE_CONFLICTMONITOR` then the CM topologies are enabled.

It might make sense to use different environmental variables, but having someway to enable/disable based off of environmental variables seems necessary before it can be part of `jpo-utils`

**Comment by John-Wiens** [comment]
> Can we configure these topologies to start based off of the same/similar environmental variables that the kafka-setup pod uses?
> 
> if `KAFKA_TOPIC_CREATE_ODE` then the raw ODE topologies are enabled. if `KAFKA_TOPIC_CREATE_GEOJSONCONVERTER` then the Processed topologies are enabled. if `KAFKA_TOPIC_CREATE_CONFLICTMONITOR` then the CM topologies are enabled.
> 
> It might make sense to use different environmental variables, but having someway to enable/disable based off of environmental variables seems necessary before it can be part of `jpo-utils`


I added a few new environment variables to the sample.env file. 
ENABLE_PROCESSED_MAP_DEDUPLICATION
ENABLE_PROCESSED_MAP_WKT_DEDUPLICATION
ENABLE_ODE_MAP_DEDUPLICATION
ENABLE_ODE_TIM_DEDUPLICATION
ENABLE_ODE_RAW_ENCODED_TIM_DEDUPLICATION
ENABLE_PROCESSED_SPAT_DEDUPLICATION
ENABLE_ODE_BSM_DEDUPLICATION

Each of these variables is by default tied to the corresponding KAFKA_TOPIC_CREATE variable. For example, the ENABLE_PROCESSED_MAP_DEDUPLICATION is set to the value of KAFKA_TOPIC_CREATE_GEOJSONCONVERTER by default. This means that if the user is using the geojson converter, the corresponding deduplication topics will also be activated. Alternatively, users may manually configure each of these variables in their .env file.

File: docker-compose-deduplicator.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    environment:
+      DOCKER_HOST_IP: ${DOCKER_HOST_IP}
+      KAFKA_BOOTSTRAP_SERVERS: ${KAFKA_BOOTSTRAP_SERVERS:?error}
+      spring.kafka.bootstrap-servers: ${KAFKA_BOOTSTRAP_SERVERS:?error}
+      enableProcessedMapDeduplication: ${ENABLE_PROCESSED_MAP_DEDUPLICATION}
+      enableProcessedMapWktDeduplication: ${ENABLE_PROCESSED_MAP_WKT_DEDUPLICATION}
+      enableOdeMapDeduplication: ${ENABLE_ODE_MAP_DEDUPLICATION}
+      enableOdeTimDeduplication: ${ENABLE_ODE_TIM_DEDUPLICATION}
+      enableOdeRawEncodedTimDeduplication: ${ENABLE_ODE_RAW_ENCODED_TIM_DEDUPLICATION}
+      enableProcessedSpatDeduplication: ${KAFKA_TOPIC_CREATE_GEOJSONCONVERTER}
```
Comment (Michael7371) [inline docker-compose-deduplicator.yml]: This should be referencing: ${ENABLE_PROCESSED_SPAT_DEDUPLICATION}

File: sample.env [hunk 2]
```diff
@@ -2,6 +2,9 @@
 # (Required) The IP address of Docker host machine which can be found by running "ifconfig"
 # Hint: look for "inet addr:" within "eth0" or "en0" for OSX
 DOCKER_HOST_IP=""
+MAVEN_GITHUB_TOKEN=
```
Comment (Michael7371) [inline sample.env]: Can we link to documentation on how to generate this in the sample.env. Something like this:
```.env
# Required for building the deduplicator. Documentation: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages
```
Also can you add a section to the `readme.md` that documents the environmental variables and background of the deduplicator application?

Comment (Michael7371) [reply sample.env]: Also should this be in the deduplicator portion of the `sample.env` as it is the only component that uses it?

**Review by Michael7371 — COMMENTED** [review]
Just a few comments. I was able to build the deduplicator and verify that it starts up correctly

File: docker-compose-deduplicator.yml (hunk 1 above)
Comment (John-Wiens) [reply docker-compose-deduplicator.yml]: Fixed!

File: sample.env (hunk 2 above)
Comment (John-Wiens) [reply sample.env]: This could be shared in the future, but for now its fine to put in the deduplicator section. Fixed.

**Review by Michael7371 — APPROVED** [review]
LGTM!

**Pull request merged**